ext/pdo_mysql/tests/bug80458.phpt

```diff
@@ -0,0 +1,156 @@
+--TEST--
+Bug #80458 PDOStatement::fetchAll() throws for upsert queries
+--SKIPIF--
+<?php
+if (!extension_loaded('pdo') || !extension_loaded('pdo_mysql')) die('skip not loaded');
+require_once(__DIR__ . DIRECTORY_SEPARATOR . 'skipif.inc');
+require_once(__DIR__ . DIRECTORY_SEPARATOR . 'mysql_pdo_test.inc');
+MySQLPDOTest::skip();
+?>
+--FILE--
+<?php
+require_once(__DIR__ . DIRECTORY_SEPARATOR . 'mysql_pdo_test.inc');
+
+$db = MySQLPDOTest::factory();
+$db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+$db->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
+
+$db->query('DROP TABLE IF EXISTS test');
+$db->query('CREATE TABLE test (first int) ENGINE = InnoDB');
+$res = $db->query('INSERT INTO test(first) VALUES (1),(2),(3),(4),(5),(6),(7),(8),(9),(10),(11),(12),(13),(14)');
+var_dump($res->fetchAll());
+
+$stmt = $db->prepare('DELETE FROM test WHERE first=1');
+$stmt->execute();
+var_dump($stmt->fetchAll());
+
+$res = $db->query('DELETE FROM test WHERE first=2');
+var_dump($res->fetchAll());
+
+$stmt2 = $db->prepare('DELETE FROM test WHERE first=3');
+$stmt2->execute();
+foreach($stmt2 as $row){
+    // expect nothing
+}
+
+$stmt3 = $db->prepare('DELETE FROM test WHERE first=4');
+$stmt3->execute();
+var_dump($stmt3->fetch(PDO::FETCH_ASSOC));
+
+$stmt = $db->prepare('SELECT first FROM test WHERE first=5');
+$stmt->execute();
+var_dump($stmt->fetchAll());
+
+$db->exec('DROP PROCEDURE IF EXISTS empty');
+$db->exec('CREATE PROCEDURE empty() BEGIN DELETE FROM test WHERE first=6; END;');
+$stmt4 = $db->prepare('CALL empty()');
+$stmt4->execute();
+var_dump($stmt4->fetchAll());
+$db->exec('DROP PROCEDURE IF EXISTS empty');
+
+$db->exec('DROP PROCEDURE IF EXISTS ret');
+$db->exec('CREATE PROCEDURE ret() BEGIN SELECT first FROM test WHERE first=7; END;');
+$stmt5 = $db->prepare('CALL ret()');
+$stmt5->execute();
+var_dump($stmt5->fetchAll());
+$stmt5->nextRowset(); // needed to fetch the empty result set of CALL
```

Comment:
Should this also do a fetchAll() after the nextRowset()? If I'm understanding the interpretation we're going with right, then that should also work.

Reply:
Yes, it should. Added.